powerline/segments/common/bat.py

```diff
@@ -46,39 +47,62 @@ def _fetch_battery_info(pl):
 					if not bool(devget('PowerSupply')):
 						pl.debug('Not using DBUS+UPower with {0}: not a power supply', devpath)
 						continue
+					devices.append(devpath)
 					pl.debug('Using DBUS+UPower with {0}', devpath)
-					return lambda pl: (
-						float(
-							dbus.Interface(dev, dbus_interface=devinterface).Get(
+				if devices:
+					def _flatten_battery(pl):
+						energy = 0.0
+						energy_full = 0.0
+						state = True
+						for devpath in devices:
+							dev = bus.get_object(interface, devpath)
+							energy_full += float(
+								dbus.Interface(dev, dbus_interface=devinterface).Get(
+									devtype_name,
+									'EnergyFull'
+								),
+							)
+							energy += float(
+								dbus.Interface(dev, dbus_interface=devinterface).Get(
+									devtype_name,
+									'Energy'
+								),
+							)
+							state &= dbus.Interface(dev, dbus_interface=devinterface).Get(
 								devtype_name,
-								'Percentage'
-							),
-						),
-						dbus.Interface(dev, dbus_interface=devinterface).Get(
-							devtype_name,
-							'State'
-						) != 2
-					)
+								'State'
+							) != 2
+						return (energy * 100.0 / energy_full), state
+					return _flatten_battery
 				pl.debug('Not using DBUS+UPower as no batteries were found')
 
 	if os.path.isdir('/sys/class/power_supply'):
-		linux_capacity_fmt = '/sys/class/power_supply/{0}/capacity'
+		linux_energy_full_fmt = '/sys/class/power_supply/{0}/enery_full'
```

Comment:
Ah! You are completely right... I fixed the filenames in the most recent commit.